### PR TITLE
Frontend fixes and python integration 

### PR DESCRIPTION
**Pull request by rryoung98**
# Changes
- Updated the task flow to start with making a template. 
- When working on plaquettes, users can now only work within a single template. 
- Moved functions out of the main index.js file into respective classes

## Review Requests
- The code requires accessing different containers in several classes and having some feedback on what a sound software architecture would be helpful. 
Ex: We are facing issues which are in-line with what react-redux solves. The TemplateClass now houses the plaquette generation and creates plaquette classes which are added into the templateClass container. However, we may need to conduct a lot of prop drilling and assigning the app as a global variable may make interfacing with the canvas a lot easier and remove the necessity to drill prop.

## Next Steps
- Add a done button for when the user is done making a template.
- Add top left qubit indexing to Template
- Add multi-template functionality (nice-to-have, not required)
- Add circuit visualizer #46
- Once circuit visualizer is completed, connect with the download stim file button.

https://github.com/QCHackers/tqec/assets/32727721/d187d960-570d-4f8d-a623-35427f0aa396

### CODE REVIEW

**Review by nelimee — COMMENTED** [review]
Not sure you meant that, but I do not think the changes to the `notebook` directory should be merged. Did I miss something?

**Comment by rryoung98** [comment]
> Not sure you meant that, but I do not think the changes to the `notebook` directory should be merged. Did I miss something?

@nelimee, yes I just used nbstripout to remove the output in the notebook. I can revet it

**Comment by nelimee** [comment]
> > Not sure you meant that, but I do not think the changes to the `notebook` directory should be merged. Did I miss something?
> 
> @nelimee, yes I just used nbstripout to remove the output in the notebook. I can revert it

If possible, yes. I included the output on purpose so that people do not have to re-run everything to see the effect of the code. The notebooks are basically a showcase, a documentation, and also functional tests for the moment, so the output is important and is, in my opinion, worth appearing on the GitHub repository.

File: frontend/src/workspace/TemplateClass.js [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+			notification(this.app, 'Step 3: Click on 3+ qubits to make a plaquette');
+			this.app.view.addEventListener('click', this.selectQubit);
+			this.plaquetteButton.visible = true;
+			this.plaquetteButton.on('click', (_e) => {
+				// Create the plaquettes and tile
+				this.createPlaquette();
+				this.workspace.addChild(this.container);
+				// Clear the selected qubits
+				this.selectedQubits = [];
+				notification(this.app, 'Step 4: Define the circuit');
```
Comment (smburdick) [inline frontend/src/workspace/TemplateClass.js:110]: Need to make an issue for defining circuits.

**Review by smburdick — APPROVED** [review]
LGTM. This is exactly how I envisioned template creation.

* Since you performed linting in https://github.com/QCHackers/tqec/pull/69/commits/5fc58d85813f0efa66e7a79a2ca56d10091a9d4b, do we want to define or add auto-lint for this project?
* Some build warnings for posterity --
```js
Compiled with warnings.

[eslint]
src/workspace/TemplateClass.js
  Line 153:43:  Array.prototype.filter() expects a value to be returned at the end of arrow function  array-callback-return

src/workspace/plaquettes/PlaquetteClass.js
  Line 3:8:  'notification' is defined but never used  no-unused-vars

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in [eslint]
src/workspace/TemplateClass.js
  Line 153:43:  Array.prototype.filter() expects a value to be returned at the end of arrow function  array-callback-return

src/workspace/plaquettes/PlaquetteClass.js
  Line 3:8:  'notification' is defined but never used  no-unused-vars

webpack compiled with 1 warning
```

**Comment by rryoung98** [comment]
Thanks @smburdick yes auto listing would be really good

**Comment by rryoung98** [comment]
I can stick a prettier and eslint in this commit

**Comment by smburdick** [comment]
> I can stick a prettier and eslint in this commit

Are we ready to merge?

**Comment by rryoung98** [comment]
all right we're all set to merge

**Pull request merged**